app/test/testattributecontroller.cpp

```diff
@@ -718,30 +725,186 @@ void TestAttributeController::testPhotoRenaming()
 
   QVERIFY( surveyLayer && surveyLayer->isValid() );
 
-  QgsFeature feat;
-  feat.setValid( true );
-  feat.setFields( surveyLayer->fields(), true );
+  QgsFeature feat( surveyLayer->fields() );
   FeatureLayerPair pair( feat, surveyLayer );
 
   AttributeController controller;
   controller.setFeatureLayerPair( pair );
 
   const TabItem *tab = controller.tabItem( 0 );
   const QVector<QUuid> items = tab->formItems();
-  QCOMPARE( items.size(), 5 );
+  QCOMPARE( items.size(), 9 );
+
+  //
+  // Attachment fields:
+  // 3 - photo
+  // 4 - photo_with_slash
+  // 5 - photo_in_subfolder_relative_to_project_path -> subfolder 'media'
+  // 6 - photo_in_subfolder_relative_to_default_path -> subfolder 'media'
+  // 7 - photo_with_slash_in_subfolder_relative_to_project_path -> subfolder 'media/with_slash'
+  // 8 - photo_with_slash_in_subfolder_relative_to_default_path -> subfolder 'media/with_slash'
+  //
+
+  controller.setFormValue( items.at( 2 ), QStringLiteral( "mynotes" ) );
```

Comment:
How about testing the trimming too?
```suggestion
  controller.setFormValue( items.at( 2 ), QStringLiteral( "mynotes " ) );
```